modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -373,24 +373,38 @@ private Predicate<T> _getGTPredicate(
 	private Predicate<T> _getINPredicate(
 		EntityField entityField, List<Object> fieldValues) {
 
-		Stream<Object> stream = fieldValues.stream();
+		return p -> {
+			for (Object fieldValue : fieldValues) {
+				if (StringUtils.containsIgnoreCase(
+						String.valueOf(p.get(entityField.getName())),
+						String.valueOf(fieldValue))) {
 
-		return p -> stream.anyMatch(
-			fieldValue -> StringUtils.containsIgnoreCase(
-				String.valueOf(p.get(entityField.getName())),
-				String.valueOf(fieldValue)));
+					return true;
+				}
+			}
+
+			return false;
+		};
 	}
 
 	private Predicate<T> _getLambdaContainsPredicate(
 		EntityField entityField, Object fieldValue) {
 
 		if (Objects.equals(entityField.getType(), EntityField.Type.STRING)) {
-			return p -> Stream.of(
-				(String[])p.get(_lambdaCollectionEntityField.getName())
-			).anyMatch(
-				c -> StringUtils.containsIgnoreCase(
-					String.valueOf(c), String.valueOf(fieldValue))
-			);
+			return p -> {
```

Comment:
@jeffwu0724 Please flip this if-condition. An easy way is to press Alt-Enter when your IntelliJ IDEA editor cursor is on the `if` and select "Flip if condition", and then auto SF.